draft-saml-ipsie-sl1-profile.md

```diff
@@ -0,0 +1,382 @@
+
+#### 3.1.3. Requirements for endpoints used by web browsers
+
+For endpoints that are used by web browsers, the following additional requirements apply:
+
+- Servers shall use methods to ensure that connections cannot be downgraded using TLS stripping attacks. A preloaded HTTP Strict Transport Security policy [RFC6797] shall be used for this purpose.
+- When using TLS 1.2, servers shall only use cipher suites allowed in [BCP195].
+- Servers shall not support CORS for the SSO endpoints, as clients must perform an HTTP redirect rather than access these endpoints directly.
+
+### 3.2. Cryptography and Signatures
+
+The following requirements apply to cryptographic operations and signatures:
+
+- Identity Providers and Service Providers when creating or processing SAML assertions shall:
+  - adhere to XML Signature best practices [XMLDSig];
+  - use RSA-SHA256, ECDSA-SHA256, or EdDSA (using the Ed25519 variant) algorithms for XML signatures;
+  - not use unencrypted or unsigned assertions under any circumstances.
+
+- RSA keys shall have a minimum length of 2048 bits.
+- Elliptic curve keys shall have a minimum length of 256 bits.
+- Credentials not intended for handling by end-users shall be created with at least 128 bits of entropy such that an attacker correctly guessing the value is computationally infeasible.
+- All key material shall be generated using FIPS 140-2 Level 1 validated random bit generators or higher.
+- Assertion signing keys shall be distinct from keys used for other purposes.
+
+### 3.3. SAML Requirements
+
+In the following, a profile of the following technologies is defined:
+
+- SAML 2.0 Core [SAML2Core]
+- SAML 2.0 Profiles [SAML2Profiles]
+- SAML 2.0 Metadata [SAML2Metadata]
+- SAML 2.0 Bindings [SAML2Bindings]
+
+#### 3.3.1. Requirements for Identity Providers
+
+Identity Providers:
+
+- shall distribute metadata as specified in [SAML2Metadata] through a protected channel with integrity protection;
+- shall only support pre-registered Service Providers with established trust relationships;
+- shall authenticate Service Providers using X.509 certificates or another strong authentication method;
+- shall sign all assertions using XML Signature [XMLDSig];
+- shall encrypt all assertions containing personally identifiable information (PII) or sensitive attributes using XML Encryption [XMLEnc] with AES-GCM with a minimum key length of 128 bits;
+- shall not expose open redirectors;
+- shall only accept entity IDs registered in metadata as audience values;
+- shall issue assertions with a maximum lifetime of 5 minutes;
+- shall require Service Providers to be preregistered, and shall not support dynamic registration of Service Providers;
+- shall support assertion revocation or status verification mechanisms (such as OCSP for certificate checking);
+- shall implement key rotation procedures with a maximum key lifetime of 2 years for signing keys.
+
+SAML Assertions issued by Identity Providers:
```

Comment:
Some additional items would be helpful to define are:
- Specify where the signature is required (SAMLResponse, Assertion, or Both).   I prefer SAMLResponse as it makes it easier to fail fast
- Specify that there should be only 1 assertion element per response with a single subject
- Specify what name identifier formats must be supported
- Specify whether `Destination` and/or `Recipient` is required and what values should be